ONSdigital/dp-dd-api
# Feature/segregate int tests

What has changed?
Split unit and int tests into 2 groups defined by annotations in the test classes.
Unit and int have their own testng.xml config files which are invoked by their respective Scala class.

How to test?

sbt "test-only scala.UnitTestRunner"
and
sbt "test-only scala.IntTestRunner"

More detail:

From sbt Reference Manual: "The tasks for running tests are test and testOnly. The test task accepts no command line arguments and runs all tests."
Couldn't find a way to exclude int tests on a 'sbt test' command.

Unit and int tests have been grouped and given separate testng.xml config files and can be executed via their respective Scala wrapper class:
sbt "test-only scala.UnitTestRunner"
or
sbt "test-only scala.IntTestRunner"

The build process should be modified to invoke the UnitTestRunner as the default 'sbt test' will always run all tests.
